### PR TITLE
Fix  backend rest api issues of search

### PR DESCRIPTION
**Pull request by TanyaM**
This PR provides the ability to search as "Bank API" vs Bank API
- "Bank API" --> search for exact string Bank API
- Bank API --> search for either Bank or API

### CODE REVIEW

**Comment by bhathiya** [comment]
Do we support "bank" and "api"?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -7869,7 +7869,8 @@ public static String constructNewSearchQuery(String query) throws APIManagementE
 
         // sub context and doc content doesn't support AND search
         if (inputSearchQuery != null && inputSearchQuery.contains(" ") && !inputSearchQuery
-                .contains(APIConstants.TAG_SEARCH_TYPE_PREFIX4)) {
+                .contains(APIConstants.TAG_SEARCH_TYPE_PREFIX4) && (!inputSearchQuery
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: shall we rename this constant to `TAG_COLON_SEARCH_TYPE_PREFIX`? 
we can rename other similar constants too similar way. 
TAGS_SEARCH_TYPE_PREFIX
TAG_EQ_SEARCH_TYPE_PREFIX etc.

Comment (TanyaM) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Done

**Comment by TanyaM** [comment]
> Do we support "bank" and "api"?

No we don't support. AND search is not supported.

**Pull request merged**